packages/loader/container-loader/src/retriableDocumentStorageService.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { CreateContainerError } from "@fluidframework/container-utils";
+import { IDocumentStorageService } from "@fluidframework/driver-definitions";
+import { canRetryOnError, DocumentStorageServiceProxy } from "@fluidframework/driver-utils";
+import { ISnapshotTree, IVersion } from "@fluidframework/protocol-definitions";
+import { Container } from "./container";
+import { emitThrottlingWarning, getRetryDelayFromError } from "./deltaManager";
+
+export class RetriableDocumentStorageService extends DocumentStorageServiceProxy {
+    private shouldRetry = true;
+    constructor(
+        internalStorageService: IDocumentStorageService,
+        private readonly container: Container,
```

Comment:
Can it be scoped to some reasonable (i.e. much smaller) interface? I.e. only what this class needs?

Reply:
Scoped for only the needed apis.